### PR TITLE
Widget depth settings and improvements

### LINKED ISSUE #16132: Allow to limit category widget by level
Considering to add a level limit to the category widget (e.g only display top level categories)

### PR DESCRIPTION
**Pull request by mikejolley**
- Standardize widget names with title case and descriptions to match WP
- Add max depth option which closes #16132 
- Remove legacy code - we can use wp_dropdown_categories again! https://core.trac.wordpress.org/ticket/13258

### CODE REVIEW

File: includes/widgets/class-wc-widget-product-categories.php [hunk 1]
```diff
@@ -101,26 +105,26 @@ public function widget( $args, $instance ) {
 		$hide_empty         = isset( $instance['hide_empty'] ) ? $instance['hide_empty'] : $this->settings['hide_empty']['std'];
 		$dropdown_args      = array( 'hide_empty' => $hide_empty );
 		$list_args          = array( 'show_count' => $count, 'hierarchical' => $hierarchical, 'taxonomy' => 'product_cat', 'hide_empty' => $hide_empty );
+		$max_depth          = absint( isset( $instance['max_depth'] ) ? $instance['max_depth'] : $this->settings['max_depth']['std']);
```
Comment (claudiulodro) [inline includes/widgets/class-wc-widget-product-categories.php]: Needs a space at the end of the line.

File: includes/widgets/class-wc-widget-product-categories.php [hunk 2]
```diff
@@ -222,17 +223,14 @@ public function widget( $args, $instance ) {
...
-
 			include_once( WC()->plugin_path() . '/includes/walkers/class-product-cat-list-walker.php' );
 
 			$list_args['walker']                     = new WC_Product_Cat_List_Walker;
 			$list_args['title_li']                   = '';
 			$list_args['pad_counts']                 = 1;
 			$list_args['show_option_none']           = __( 'No product categories exist.', 'woocommerce' );
-			$list_args['current_category']           = ( $this->current_cat ) ? $this->current_cat->term_id : '';
+			$list_args['current_category']           = ( $this->current_cat ) ? $this->current_cat->term_id: '';
 			$list_args['current_category_ancestors'] = $this->cat_ancestors;
```
Comment (claudiulodro) [inline includes/widgets/class-wc-widget-product-categories.php]: Needs a space before the `:`

**Comment by claudiulodro** [comment]
Looks good. I like the new phrasing on the widgets. Just a couple minor formatting tweaks and it's good to go.

**Comment by claudiosanches** [comment]
I like the new widget names.

**Pull request merged**